plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiUtil.java

```diff
@@ -0,0 +1,198 @@
+            return HoodieFileFormat.HOODIE_LOG;
+        }
+        if (extension.equals(HoodieFileFormat.ORC.getFileExtension())) {
+            return HoodieFileFormat.ORC;
+        }
+        if (extension.equals(HoodieFileFormat.HFILE.getFileExtension())) {
+            return HoodieFileFormat.HFILE;
+        }
+        throw new HoodieIOException("Hoodie InputFormat not implemented for base file of type " + extension);
+    }
+
+    public static boolean partitionMatchesPredicates(
+            SchemaTableName tableName,
+            String hivePartitionName,
+            List<HiveColumnHandle> partitionColumnHandles,
+            TupleDomain<HiveColumnHandle> constraintSummary)
+    {
+        List<Type> partitionColumnTypes = partitionColumnHandles.stream()
+                .map(HiveColumnHandle::getType)
+                .collect(toList());
+        HivePartition partition = HivePartitionManager.parsePartition(
+                tableName, hivePartitionName, partitionColumnHandles, partitionColumnTypes);
+
+        return partitionMatches(partitionColumnHandles, constraintSummary, partition);
+    }
+
+    public static boolean partitionMatchesPredicates(
+            SchemaTableName tableName,
+            String relativePartitionPath,
+            List<String> partitionValues,
+            List<HiveColumnHandle> partitionColumnHandles,
+            TupleDomain<HiveColumnHandle> constraintSummary)
+    {
+        List<Type> partitionColumnTypes = partitionColumnHandles.stream()
+                .map(HiveColumnHandle::getType)
+                .collect(toList());
+        HivePartition partition = parsePartition(
+                tableName, relativePartitionPath, partitionValues, partitionColumnHandles, partitionColumnTypes);
+
+        return partitionMatches(partitionColumnHandles, constraintSummary, partition);
+    }
+
+    private static HivePartition parsePartition(
+            SchemaTableName tableName,
+            String partitionName,
+            List<String> partitionValues,
+            List<HiveColumnHandle> partitionColumns,
+            List<Type> partitionColumnTypes)
+    {
+        ImmutableMap.Builder<ColumnHandle, NullableValue> builder = ImmutableMap.builder();
+        for (int i = 0; i < partitionColumns.size(); i++) {
+            HiveColumnHandle column = partitionColumns.get(i);
+            NullableValue parsedValue = parsePartitionValue(
+                    partitionName, partitionValues.get(i), partitionColumnTypes.get(i));
+            builder.put(column, parsedValue);
+        }
+        Map<ColumnHandle, NullableValue> values = builder.buildOrThrow();
+        return new HivePartition(tableName, partitionName, values);
+    }
+
+    public static boolean partitionMatches(List<HiveColumnHandle> partitionColumns, TupleDomain<HiveColumnHandle> constraintSummary, HivePartition partition)
+    {
+        if (constraintSummary.isNone()) {
+            log.warn("constraintSummary is none");
```

Comment:
Is this warning useful for an administrator? If so, make it actionable by including table name, etc., so that they could do something about it like cleaning up the metastore. Otherwise, remove it.

Reply:
it's not very useful. will remove it.